embulk-core/src/main/java/org/embulk/spi/ExecSession.java

```diff
@@ -219,6 +233,16 @@ public boolean isPreview()
 
     public void cleanup()
     {
+        for (final Reporter.Channel channel : Reporter.Channel.values()) {
+            final AbstractReporterImpl reporter = (AbstractReporterImpl) reporters.get(channel);
```

Comment:
Yes, `reporters` is null.. As commented above, `GuessExecutor` or `PreviewExecutor` didn't set `reporters` to the session. I will care of them.